### ISSUE TITLE
Enable eslint-plugin-jsx-a11y and fix existing issues

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#1727]
As part of our general effort to improve accessibility and achieve WCAG compliance, introduce static linting to check for a11y errors in markup.

- [x] Enable the jsx-a11y ESLint plugin for the client. Any existing failures should be flagged with a comment (`FIXME-A11Y`) and then temporarily suppressed (https://github.com/hypothesis/client/pull/1728)
- [x] Fix errors in `annotation-header.js` (#1782)
- [x] Fix errors in `excerpt.js`
- [x] Fix errors in `group-list-item.js`
- [x] Fix errors in `group-list.js`
- [x] Fix errors in `help-panel.js`
- [x] Fix errors in `logged-out-message.js`
- [x] Fix errors in `selection-tabs.js`
- [x] Fix errors in `sidebar-content-error.js` (#1791)
- [x] Fix errors in `top-bar.js` https://github.com/hypothesis/client/pull/1784

_(Menu/MenuItem) is pending discussion https://hypothes-is.slack.com/archives/C1M8NH76X/p1581402527176000_

- [x] Fix errors in `menu-item.js` (in prog -- kyle) (superceded by https://github.com/hypothesis/client/issues/1735 - @robertknight )
- [x] Fix errors in `menu.js` (in prog -- kyle) (superceded by https://github.com/hypothesis/client/issues/1735 - @robertknight )

**Comment by robertknight** [comment]
The jsx-a11y issues have been addressed except for the menus, which need more work to implement the standard keyboard interaction and focus patterns. This is captured in https://github.com/hypothesis/client/issues/1735.
